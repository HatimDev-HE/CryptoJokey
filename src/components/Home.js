import React, { useEffect, useState } from 'react';
import { NavLink } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { getCrypto } from '../redux/API/fruitsfetch';
import View from './Assets/view.svg';

const Home = () => {
  const crypto = useSelector((state) => state.crypto);
  const dispatch = useDispatch();
  useEffect(() => {
    if (crypto.length === 0) {
      dispatch(getCrypto());
    }
  }, [dispatch]);
  const [query, setQuery] = useState('');
  return (
    <div>
      <div className="headall">
        <input type="search" id="search" name="search" placeholder="Search Jokey" onChange={(event) => setQuery(event.target.value)} />
        <p className="jokey-coins">All Jokey Coins</p>
      </div>
      <>
        {crypto
          .filter((crypto) => {
            if (query === '') {
              return true;
            }
            if (crypto.name.toLowerCase().includes(query.toLowerCase())) {
              return true;
            }
            return false;
          })
          .map((crypto) => (
            <div className="cryptos" key={crypto.name}>
              <div className="namerow">
                <li className="name">{crypto.name}</li>
                <li className="change">
                  Price Change:
                  {' '}
                  {crypto.priceChange}
                </li>
                <li className="change">
                  Bid Price:
                  {' '}
                  {crypto.bidPrice}
                </li>
                <NavLink
                  className="align-self-center arrow"
                  to="/details"
                  state={crypto}
                >
                  <p className="details">
                    <li>
                      <img src={View} alt="" height={30} width={30} className="detailimg" />
                    </li>
                  </p>
                </NavLink>

              </div>
            </div>
          ))}
      </>
    </div>
  );
};

export default Home;
