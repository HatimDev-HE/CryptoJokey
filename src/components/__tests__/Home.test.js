import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/configstore';
import Home from '../Home';

describe('testing Home comp', () => {
  test('Home to match snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
