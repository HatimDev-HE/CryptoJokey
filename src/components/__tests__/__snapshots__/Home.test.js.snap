// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home comp Home to match snapshot 1`] = `
<div>
  <div
    className="headall"
  >
    <input
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="Search Jokey"
      type="search"
    />
    <p
      className="jokey-coins"
    >
      All Jokey Coins
    </p>
  </div>
</div>
`;

exports[`testing Nav comp Home to match snapshot 1`] = `
<div>
  <div
    className="headall"
  >
    <input
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="Search Jokey"
      type="search"
    />
    <p
      className="jokey-coins"
    >
      All Jokey Coins
    </p>
  </div>
</div>
`;

exports[`testing Nav comp Nav to match snapshot 1`] = `
<div>
  <div
    className="headall"
  >
    <input
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="Search Jokey"
      type="search"
    />
    <p
      className="jokey-coins"
    >
      All Jokey Coins
    </p>
  </div>
</div>
`;
